lib/vehicle_factory.rb

```diff
@@ -0,0 +1,50 @@
+class VehicleFactory
+  attr_reader :vehicle_instances
+
+  def initialize
+    @vehicle_instances = []
+  end
```

Comment:
Similar to your FacilityFactory.. no need for you to store state for the vehicle objects.